src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1033,6 +1039,21 @@ public StringBuilder Append(char[] value)
             return this;
         }
 
+        public StringBuilder Append(ReadOnlySpan<char> value)
+        {
+            if (value.Length > 0)
+            {
+                unsafe
+                {
+                    fixed (char* valueChars = &value.DangerousGetPinnableReference())
```

Comment:
Since there's not currently interop support for Span (@yizhang82, any plans?), I agree there are places where pointers are still necessary. However, if we minimize those places, it mitigates cases like the int overflows you pointed out.